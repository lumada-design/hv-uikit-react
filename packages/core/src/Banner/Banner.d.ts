import * as React from "react";
import { StandardProps, SnackbarProps, SnackbarContentProps } from "@material-ui/core";
import { HvActionsGenericCommonProps } from "../ActionsGeneric";

export type TransitionDirectionProp = "up" | "down" | "left" | "right";

export type ActionsPositionProp = "auto" | "inline" | "bottom-right";

export type SemanticVariantTypes = "success" | "warning" | "error" | "info" | "default";

export type HvBannerClassKey =
  | "root"
  | "rootClosed"
  | "anchorOriginTopCenter"
  | "anchorOriginBottomCenter";

/**
 * Interface shared between HvBanner and HvSnackbar
 */
export interface NotificationsCommonProps {
  /**
   * The message to display.
   */
  label?: string;
  /**
   * Variant of the snackbar.
   */
  variant?: SemanticVariantTypes;
  /**
   * Custom icon to replace the variant default.
   */
  customIcon?: React.ReactNode;
  /**
   * Controls if the associated icon to the variant should be shown.
   */
  showIcon?: boolean;
  /**
   * Direction of slide transition.
   */
  transitionDirection?: TransitionDirectionProp;
  /**
   * Offset from top/bottom of the page, in px. Defaults to 60px.
   */
  offset?: number;
}

export interface HvBannerProps
  extends StandardProps<SnackbarProps, HvBannerClassKey>,
    HvActionsGenericCommonProps,
    NotificationsCommonProps {
  /**
   * The position property of the header.
   */
  actionsPosition?: ActionsPositionProp;
  /**
   * Props to pass down to the Banner Wrapper.
   */
  bannerContentProps?: SnackbarContentProps;
}

export default function HvBanner(props: HvBannerProps): JSX.Element | null;
