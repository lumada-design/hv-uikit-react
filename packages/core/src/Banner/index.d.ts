declare module '@hv/uikit-react-core/dist/Banner' {
  class HvBanner extends React.Component<HvBannerProps, any> {}

  export default HvBanner

  export interface BannerAction {
    id: string
    label: string
    icon?: (...args: any[]) => any
    disabled?: boolean
  }

  export interface HvBannerProps extends React.HTMLAttributes<HvBanner> {
    /**
     * A Jss Object used to override or extend the styles applied.
     */
    classes?: {
      /**
       * Styles applied to the component root class.
       */
      root?: string
      /**
       * Styles applied to the component when define as top.
       */
      anchorOriginTopCenter?: string
      /**
       * Styles applied to the component when define as bottom.
       */
      anchorOriginBottomCenter?: string
    }

    /**
     *  If true, Snackbar is open.
     */
    open?: boolean

    /**
     * Callback fired when the component requests to be closed. Typically onClose is used to set state in the parent component, which is used to control the Snackbar open prop. The reason parameter can optionally be used to control the response to onClose, for example ignoring clickaway.
     */
    onClose?: (...args: any[]) => any

    /**
     * The message to display.
     */
    label?: string

    /**
     *  The anchor of the Snackbar.
     */
    anchorOrigin?: 'top' | 'bottom'

    /**
     * Variant of the snackbar.
     */
    variant?: 'success' | 'warning' | 'error' | 'info' | 'default'

    /**
     * Custom icon to replace the variant default.
     */
    customIcon?: React.ReactNode

    /**
     * Controls if the associated icon to the variant should be shown.
     */
    showIcon?: boolean

    /**
     * Actions to display on the right side.
     */
    actions?: React.ReactNode | BannerAction[]

    /**
     *  The callback function ran when an action is triggered, receiving ´action´ as param
     */
    actionsCallback?: (...args: any[]) => any

    /**
     * The position property of the header.
     */
    actionsPosition?: 'auto' | 'inline' | 'bottom-right'

    /**
     * How much the transition animation last in milliseconds, if 0 no animation is played.
     */
    transitionDuration?: number

    /**
     * Direction of slide transition.
     */
    transitionDirection?: 'up' | 'down' | 'left' | 'right'

    /**
     * Offset from top/bottom of the page, in px. Defaults to 60px.
     */
    offset?: number

    // deprecated:
    /**
     * The message to display.
     * @deprecated. Instead use the label property
     */
    message?: string

    /**
     * Actions to display on the right side.
     * @deprecated. Instead use the actions property
     */
    action?: React.ReactNode

    /**
     * Actions to display on message.
     * @deprecated. Instead use the actions property together with actionsPosition="inline"
     */
    actionsOnMessage?: React.ReactNode
  }
}
