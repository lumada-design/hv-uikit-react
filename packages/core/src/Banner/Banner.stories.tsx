import { useState } from "react";
import styled from "@emotion/styled";
import { Meta, StoryObj } from "@storybook/react";
import { css } from "@emotion/css";
import { Info } from "@hitachivantara/uikit-react-icons";
import {
  HvBanner,
  HvBannerProps,
  HvBannerContent,
  HvBannerContentProps,
  HvButton,
  HvTypography,
  theme,
} from "@hitachivantara/uikit-react-core";

const StyledBanner = styled(HvBanner)({
  position: "relative",
  top: "0px",
  marginBottom: 10,
});
const meta: Meta<typeof HvBanner> = {
  title: "Components/Banner",
  component: HvBanner,
};
export default meta;

export const Main: StoryObj<HvBannerProps> = {
  args: {
    open: true,
    variant: "default",
    label: "This is an informational message.",
    showIcon: true,
    offset: 0,
  },
  argTypes: {
    classes: { control: { disable: true } },
    customIcon: { control: { disable: true } },
    bannerContentProps: { control: { disable: true } },
    actions: { control: { disable: true } },
    actionsCallback: { control: { disable: true } },
  },
  render: (args) => {
    return <StyledBanner {...args} />;
  },
};

export const Variants: StoryObj<HvBannerProps> = {
  parameters: {
    docs: {
      description: {
        story:
          "Banner with custom icon with base color on the semantic background.",
      },
    },
  },
  render: () => {
    const actionArray = (id: string) => [
      { id: `action${id}_1`, label: "Action 1", disabled: false },
      { id: `action${id}_2`, label: "Action 2", disabled: false },
    ];

    const BannerContent = (props: HvBannerContentProps) => (
      <>
        <br />
        <HvBannerContent {...props} />
      </>
    );

    return (
      <div>
        <HvTypography variant="title3">Semantics</HvTypography>
        <BannerContent content="This is a default banner." variant="default" />
        <BannerContent
          content="This is a success banner."
          variant="success"
          showIcon
        />
        <BannerContent
          content="This is a warning banner."
          variant="warning"
          showIcon
        />
        <BannerContent
          content="This is an error banner."
          variant="error"
          showIcon
        />
        <br />
        <HvTypography variant="title3">Actions</HvTypography>
        <br />
        <HvBanner
          open
          offset={0}
          label="This is a banner."
          showIcon
          actions={[{ id: "post", label: "Action", disabled: false }]}
          onAction={(event, action) => console.log("Clicked", action)}
          className={css({ position: "relative", top: 0 })}
        />
        <BannerContent
          content="This is a banner with two actions."
          actions={actionArray("banner3")}
          actionsPosition="inline"
        />
        <BannerContent
          content="This could be a one-line message text string with two actions on a tablet or on a desktop. However, this is actually is a two-lines message text string with two actions on a tablet or on a desktop."
          actions={actionArray("banner1")}
          actionsPosition="bottom-right"
        />
        <BannerContent
          content="This could be a one-line message text string with two actions on a tablet or on a desktop. This could be a two-lines message text string with two actions on a tablet or on a desktop. However, this is actually a three-lines message text string with two actions on a tablet or on a desktop."
          actions={actionArray("banner2")}
          actionsPosition="bottom-right"
        />
        <br />
        <HvTypography variant="title3">Custom icon</HvTypography>
        <BannerContent
          content="This is a banner with a custom icon."
          customIcon={<Info color="base_dark" />}
        />
        <BannerContent
          content="This could be a one-line message text string with no actions on a tablet or on a desktop. This could be a two-lines message text string with no actions on a tablet or on a desktop. However, this is actually a three-lines message text string with no actions on a tablet or on a desktop."
          customIcon={<Info color="base_dark" />}
        />
      </div>
    );
  },
};

export const BannerController: StoryObj<HvBannerProps> = {
  parameters: {
    docs: {
      description: {
        story:
          "Different Banner variants with actions and controlled by buttons.",
      },
    },
    eyes: { include: false },
  },
  render: () => {
    const SimpleBanner = ({
      variant,
      ...others
    }: Omit<HvBannerProps, "open">) => {
      const [open, setOpen] = useState(false);

      const handleOpen = () => setOpen(true);

      const handleClose = () => setOpen(false);

      return (
        <>
          <HvButton
            onClick={handleOpen}
            color="primary"
            style={{ width: "150px", textTransform: "capitalize", margin: 10 }}
          >
            {variant}
          </HvButton>
          <HvBanner
            open={open}
            onClose={handleClose}
            offset={10}
            variant={variant}
            showIcon
            actions={
              <HvButton
                variant="secondaryGhost"
                style={{ color: theme.colors.base_dark }}
              >
                Action
              </HvButton>
            }
            bannerContentProps={{
              actionProps: { "aria-label": "Close the banner" },
            }}
            {...others}
          />
        </>
      );
    };

    return (
      <>
        <SimpleBanner variant="default" label="This is a banner." />
        <SimpleBanner variant="success" label="This is a success banner." />
        <SimpleBanner variant="error" label="This is an error banner." />
      </>
    );
  },
};
