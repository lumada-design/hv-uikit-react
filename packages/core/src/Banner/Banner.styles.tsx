import { theme } from "@hitachivantara/uikit-styles";

import { createClasses } from "@core/utils/classes";

export const { useClasses, staticClasses } = createClasses("HvBanner", {
  /** Styles applied to the component root class when the component is closed. */
  rootClosed: {
    display: "none",
  },
  /** Styles applied to the component root class. */
  root: {
    minWidth: `calc(100% - ${theme.space.sm})`,
  },
  /** Styles applied to the component when define as top. */
  anchorOriginTopCenter: {
    top: theme.space.xs,
  },
  /** Styles applied to the component when define as bottom. */
  anchorOriginBottomCenter: {
    bottom: theme.space.xs,
  },
});
