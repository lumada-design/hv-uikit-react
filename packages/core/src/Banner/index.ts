export * from "./Banner";
export * from "./BannerContent";
