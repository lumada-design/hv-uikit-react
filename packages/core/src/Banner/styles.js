const styles = (theme) => ({
  rootClosed: {
    display: "none",
  },
  root: {
    minWidth: `calc(100% - ${theme.hv.spacing.sm}px)`,
  },
  anchorOriginTopCenter: {
    top: `${theme.hv.spacing.xs}px`,
  },
  anchorOriginBottomCenter: {
    bottom: `${theme.hv.spacing.xs}px`,
  },
});

export default styles;
