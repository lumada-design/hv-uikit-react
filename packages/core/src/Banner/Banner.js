import React, { useCallback } from "react";
import PropTypes, { oneOfType } from "prop-types";
import { Slide, Snackbar } from "@mui/material";
import { withStyles } from "@mui/styles";
import capitalize from "lodash/capitalize";
import { setId } from "../utils";
import HvBannerContentWrapper from "./BannerWrapper";
import styles from "./styles";

/**
 * A Banner displays an important and succinct message. It can also provide actions for the user to address, or dismiss.
 * It requires a user action, for it to be dismissed. Banners should appear at the top of the screen, below a top app bar.
 */
const HvBanner = (props) => {
  const {
    id,
    classes,
    className,
    open,
    onClose,
    anchorOrigin = "top",
    variant = "default",
    transitionDuration = 300,
    transitionDirection = "down",
    showIcon = false,
    customIcon,
    actions,
    actionsCallback,
    actionsPosition = "auto",
    label,
    offset = 60,
    bannerContentProps,
    ...others
  } = props;

  const anchorOriginOffset = {
    anchorOriginTop: {
      top: `${offset || 0}px`,
    },
    anchorOriginBottom: {
      bottom: `${offset || 0}px`,
    },
  };

  const anchorOriginBanner = { horizontal: "center", vertical: anchorOrigin };

  const SlideTransition = useCallback(
    (properties) => <Slide {...properties} direction={transitionDirection} />,
    [transitionDirection]
  );

  return (
    <Snackbar
      style={anchorOriginOffset[`anchorOrigin${capitalize(anchorOrigin)}`]}
      className={className}
      id={id}
      classes={{
        root: open ? classes.root : classes.rootClosed,
        anchorOriginTopCenter: classes.anchorOriginTopCenter,
        anchorOriginBottomCenter: classes.anchorOriginBottomCenter,
      }}
      anchorOrigin={anchorOriginBanner}
      TransitionComponent={SlideTransition}
      open={open}
      transitionDuration={transitionDuration}
      {...others}
    >
      <HvBannerContentWrapper
        id={setId(id, "content")}
        content={label}
        variant={variant}
        customIcon={customIcon}
        showIcon={showIcon}
        actions={actions}
        actionsCallback={actionsCallback}
        actionsPosition={actionsPosition}
        onClose={onClose}
        {...bannerContentProps}
      />
    </Snackbar>
  );
};

HvBanner.propTypes = {
  /**
   * Class names to be applied.
   */
  className: PropTypes.string,
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the component root class.
     */
    root: PropTypes.string,
    /**
     * Styles applied to the component root class when the component is closed.
     */
    rootClosed: PropTypes.string,
    /**
     * Styles applied to the component when define as top.
     */
    anchorOriginTopCenter: PropTypes.string,
    /**
     * Styles applied to the component when define as bottom.
     */
    anchorOriginBottomCenter: PropTypes.string,
  }).isRequired,
  /**
   *  If true, Snackbar is open.
   */
  open: PropTypes.bool.isRequired,
  /**
   * Callback fired when the component requests to be closed. Typically onClose is used to set state in the parent component, which is used to control the Snackbar open prop. The reason parameter can optionally be used to control the response to onClose, for example ignoring clickaway.
   */
  onClose: PropTypes.func,
  /**
   * The message to display.
   */
  label: PropTypes.node,
  /**
   *  The anchor of the Snackbar.
   */
  anchorOrigin: PropTypes.oneOf(["top", "bottom"]),
  /**
   * Variant of the snackbar.
   */
  variant: PropTypes.oneOf(["success", "warning", "error", "info", "default"]),
  /**
   * Custom icon to replace the variant default.
   */
  customIcon: PropTypes.node,
  /**
   * Controls if the associated icon to the variant should be shown.
   */
  showIcon: PropTypes.bool,
  /**
   * Actions to display on the right side.
   */
  actions: oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
        icon: PropTypes.func,
        disabled: PropTypes.bool,
      })
    ),
  ]),
  /**
   *  The callback function ran when an action is triggered, receiving `action` as param
   */
  actionsCallback: PropTypes.func,
  /**
   * The position property of the header.
   */
  actionsPosition: PropTypes.PropTypes.oneOf(["auto", "inline", "bottom-right"]),
  /**
   * How much the transition animation last in milliseconds, if 0 no animation is played.
   */
  transitionDuration: PropTypes.number,
  /**
   * Direction of slide transition.
   */
  transitionDirection: PropTypes.oneOf(["up", "down", "left", "right"]),
  /**
   * Offset from top/bottom of the page, in px. Defaults to 60px.
   */
  offset: PropTypes.number,
  /**
   * Props to pass down to the Banner Wrapper. An object `actionProps` can be included to be passed as others to actions.
   */
  bannerContentProps: PropTypes.instanceOf(Object),
};

export default withStyles(styles, { name: "HvBanner" })(HvBanner);
