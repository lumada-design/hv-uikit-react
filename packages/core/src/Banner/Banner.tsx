import { useCallback } from "react";
import Slide, { SlideProps } from "@mui/material/Slide";
import Snackbar, {
  SnackbarProps as MuiSnackbarProps,
  SnackbarOrigin,
} from "@mui/material/Snackbar";

import { HvActionGeneric } from "../ActionsGeneric";
import { ExtractNames } from "../utils/classes";
import { setId } from "../utils/setId";
import { useDefaultProps } from "../hooks/useDefaultProps";
import { useClasses, staticClasses } from "./Banner.styles";
import {
  HvBannerContent,
  HvBannerContentProps,
} from "./BannerContent/BannerContent";
import { HvBannerActionPosition, HvBannerVariant } from "./types";

export { staticClasses as bannerClasses };

export type HvBannerClasses = ExtractNames<typeof useClasses>;

export interface HvBannerProps
  extends Omit<MuiSnackbarProps, "anchorOrigin" | "classes" | "onClose"> {
  /** If true, the snackbar is open. */
  open: boolean;
  /** Callback fired when the component requests to be closed. Typically onClose is used to set state in the parent component, which is used to control the Snackbar open prop. The reason parameter can optionally be used to control the response to onClose, for example ignoring clickaway. */
  onClose?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  /** The message to display. */
  label?: string;
  /** The anchor of the Snackbar. */
  anchorOrigin?: "top" | "bottom";
  /** Variant of the snackbar. */
  variant?: HvBannerVariant;
  /** Custom icon to replace the variant default. */
  customIcon?: React.ReactNode;
  /** Controls if the associated icon to the variant should be shown. */
  showIcon?: boolean;
  /** Actions to display on the right side. */
  actions?: React.ReactNode | HvActionGeneric[];
  /** The callback function ran when an action is triggered, receiving `action` as param */
  actionsCallback?: (
    event: React.SyntheticEvent,
    id: string,
    action: HvActionGeneric
  ) => void;
  /** The position property of the header. */
  actionsPosition?: HvBannerActionPosition;
  /** How much the transition animation last in milliseconds, if 0 no animation is played. */
  transitionDuration?: number;
  /** Direction of slide transition. */
  transitionDirection?: "up" | "down" | "left" | "right";
  /** The container the banner should slide from. */
  container?: SlideProps["container"];
  /** Offset from top/bottom of the page, in px. Defaults to 60px. */
  offset?: number;
  /** Props to pass down to the banner Wrapper. An object `actionProps` can be included to be passed as others to actions. */
  bannerContentProps?: HvBannerContentProps;
  /** A Jss Object used to override or extend the styles applied to the component. */
  classes?: HvBannerClasses;
  /** @ignore */
  ref?: MuiSnackbarProps["ref"];
}

/**
 * A Banner displays an important and succinct message. It can also provide actions for the user to address, or dismiss.
 * It requires a user action, for it to be dismissed. Banners should appear at the top of the screen, below a top app bar.
 */
export const HvBanner = (props: HvBannerProps) => {
  const {
    id,
    classes: classesProp,
    className,
    open,
    onClose,
    anchorOrigin = "top",
    variant = "default",
    transitionDuration = 300,
    transitionDirection = "down",
    container,
    showIcon = false,
    customIcon,
    actions,
    actionsCallback,
    actionsPosition = "auto",
    label,
    offset = 60,
    bannerContentProps,
    ...others
  } = useDefaultProps("HvBanner", props);
  const { classes, cx } = useClasses(classesProp);

  const anchorOriginBanner: SnackbarOrigin = {
    horizontal: "center",
    vertical: anchorOrigin,
  };

  const SlideTransition = useCallback<
    NonNullable<MuiSnackbarProps["TransitionComponent"]>
  >(
    (properties) => (
      <Slide
        {...properties}
        container={container}
        direction={transitionDirection}
      />
    ),
    [container, transitionDirection]
  );

  return (
    <Snackbar
      id={id}
      open={open}
      className={className}
      classes={{
        root: cx(classes.root, { [classes.rootClosed]: !open }),
        anchorOriginTopCenter: classes.anchorOriginTopCenter,
        anchorOriginBottomCenter: classes.anchorOriginBottomCenter,
      }}
      style={{ [anchorOrigin]: offset }}
      anchorOrigin={anchorOriginBanner}
      TransitionComponent={SlideTransition}
      transitionDuration={transitionDuration}
      {...others}
    >
      <HvBannerContent
        id={setId(id, "content")}
        content={label}
        variant={variant}
        customIcon={customIcon}
        showIcon={showIcon}
        actions={actions}
        actionsCallback={actionsCallback}
        actionsPosition={actionsPosition}
        onClose={onClose}
        {...bannerContentProps}
      />
    </Snackbar>
  );
};
