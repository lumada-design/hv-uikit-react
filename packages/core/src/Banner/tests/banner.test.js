import React from "react";
import { mount } from "enzyme";
import { Fail } from "@hitachivantara/uikit-react-icons";
import { Snackbar as MaterialSnackbar } from "@material-ui/core";
import { HvButton, HvBanner, HvProvider } from "../..";

import Banner from "..";
import HvBannerContentWrapper from "../BannerWrapper";
import iconVariants from "../../utils/iconVariants";

describe("Banner ", () => {
  const wrapper = mount(
    <HvProvider>
      <HvBanner id="banner" variant="default" open={false} onClose={() => {}} />
    </HvProvider>
  );

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(Banner)).toMatchSnapshot();
  });

  it("should render the Banner component", () => {
    const bannerComponent = wrapper.find(Banner);
    expect(bannerComponent.length).toBe(1);
  });

  it("shouldn't render the BannerContentWrapper component", () => {
    const bannerComponent = wrapper.find(HvBannerContentWrapper);
    expect(bannerComponent.length).toBe(0);
  });

  it("should render the BannerContentWrapper component", () => {
    const bannerComponent = mount(
      <HvProvider>
        <HvBanner id="banner" variant="default" label="label" open onClose={() => {}} />
      </HvProvider>
    ).find(HvBannerContentWrapper);
    expect(bannerComponent.length).toBe(1);
  });

  it("should render the icon in the component", () => {
    const iconComponent = mount(
      <HvProvider>
        <HvBanner id="banner" variant="error" label="label" open showIcon onClose={() => {}} />
      </HvProvider>
    ).find(Fail);
    expect(iconComponent.length).toBe(1);
  });

  it("shouldn't render the icon in the component", () => {
    const iconComponent = mount(
      <HvProvider>
        <HvBanner
          id="banner"
          variant="error"
          label="label"
          open
          showIcon={false}
          onClose={() => {}}
        />
      </HvProvider>
    ).find(Fail);
    expect(iconComponent.length).toBe(0);
  });

  it("should render a custom icon in the component", () => {
    const iconComponent = mount(
      <HvProvider>
        <HvBanner
          id="banner"
          variant="default"
          label="label"
          open
          customIcon={<Fail />}
          onClose={() => {}}
        />
      </HvProvider>
    ).find(Fail);
    expect(iconComponent.length).toBe(1);
  });

  it("should render a action on the message", () => {
    const buttonComponent = mount(
      <HvProvider>
        <HvBanner
          id="banner"
          label="label"
          variant="default"
          open
          actions={<HvButton>button</HvButton>}
          actionsPosition="inline"
          onClose={() => {}}
        />
      </HvProvider>
    ).find(HvButton);
    expect(buttonComponent.length).toBe(1);
  });

  it("should render a action by passing a structure on the message", () => {
    const buttonComponent = mount(
      <HvProvider>
        <HvBanner
          id="banner1"
          variant="default"
          open
          label="label"
          actions={[
            {
              id: "testButton",
              label: "test",
            },
          ]}
          actionsPosition="inline"
          onClose={() => {}}
        />
      </HvProvider>
    ).find(HvButton);
    expect(buttonComponent.length).toBe(1);
  });

  it("should render a action on the action container", () => {
    const buttonComponent = mount(
      <HvProvider>
        <HvBanner
          id="banner"
          variant="default"
          open
          label="label"
          actions={<HvButton>button</HvButton>}
          actionsPosition="bottom-right"
          onClose={() => {}}
        />
      </HvProvider>
    ).find(HvButton);
    expect(buttonComponent.length).toBe(1);
  });

  it("should render a action by passing a structure on the action container", () => {
    const buttonComponent = mount(
      <HvProvider>
        <HvBanner
          id="Snackbar"
          variant="default"
          open
          label="label"
          actions={[
            {
              id: "testButton",
              label: "test",
            },
          ]}
          actionsPosition="bottom-right"
          onClose={() => {}}
        />
      </HvProvider>
    ).find(HvButton);
    expect(buttonComponent.length).toBe(1);
  });

  it("should render with the correct offset", () => {
    const offset = 10;
    let component = mount(
      <HvProvider>
        <HvBanner open offset={offset} onClose={() => {}} label="label" />
      </HvProvider>
    ).find(MaterialSnackbar);

    expect(component.get(0).props.style).toEqual({ top: `${offset}px` });

    component = mount(
      <HvProvider>
        <HvBanner open offset={offset} anchorOrigin="bottom" label="label" onClose={() => {}} />
      </HvProvider>
    ).find(MaterialSnackbar);

    expect(component.get(0).props.style).toEqual({ bottom: `${offset}px` });
  });

  it("should return the severity variant", () => {
    const error = mount(iconVariants("error")).find("Fail");
    expect(error.length).toBe(1);
    expect(iconVariants("test")).toBe(null);
  });
});
