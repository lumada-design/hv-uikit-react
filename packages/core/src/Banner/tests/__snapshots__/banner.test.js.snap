// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner  should render correctly 1`] = `
<WithStyles(HvBanner)
  id="banner"
  onClose={[Function]}
  open={false}
  variant="default"
>
  <HvBanner
    classes={
      Object {
        "anchorOriginBottomCenter": "HvBanner-anchorOriginBottomCenter",
        "anchorOriginTopCenter": "HvBanner-anchorOriginTopCenter",
        "root": "HvBanner-root",
        "rootClosed": "HvBanner-rootClosed",
      }
    }
    id="banner"
    onClose={[Function]}
    open={false}
    variant="default"
  >
    <ForwardRef(Snackbar)
      TransitionComponent={[Function]}
      anchorOrigin={
        Object {
          "horizontal": "center",
          "vertical": "top",
        }
      }
      classes={
        Object {
          "anchorOriginBottomCenter": "HvBanner-anchorOriginBottomCenter",
          "anchorOriginTopCenter": "HvBanner-anchorOriginTopCenter",
          "root": "HvBanner-rootClosed",
        }
      }
      id="banner"
      open={false}
      style={
        Object {
          "top": "60px",
        }
      }
      transitionDuration={300}
    />
  </HvBanner>
</WithStyles(HvBanner)>
`;
